packages/builders/src/web/dev-server/web-dev-server.builder.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  Builder,
+  BuildEvent,
+  BuilderConfiguration,
+  BuilderContext
+} from '@angular-devkit/architect';
+import { getSystemPath } from '@angular-devkit/core';
+import { WebpackDevServerBuilder } from '@angular-devkit/build-webpack';
+
+import { Observable } from 'rxjs';
+import { normalizeWebBuildOptions } from '../../utils/normalize';
+import { getDevServerConfig } from '../../utils/webpack/devserver.config';
+import { concatMap, map, switchMap, tap } from 'rxjs/operators';
+import { WebBuildBuilderOptions } from '../build/web-build.builder';
+import { Configuration } from 'webpack';
+import { writeFileSync } from 'fs';
+import * as opn from 'opn';
+import * as url from 'url';
+import { resolve } from 'path';
+import { buildServePath } from '../../utils/serve-path';
+import { stripIndents } from '@angular-devkit/core/src/utils/literals';
+
+export interface WebDevServerOptions {
+  host: string;
+  port: number;
+  publicHost?: string;
+  ssl: boolean;
+  sslKey?: string;
+  sslCert?: string;
+  proxyConfig?: string;
+  buildTarget: string;
+  open: boolean;
+  liveReload: boolean;
+  watch: boolean;
+}
+
+export default class BuildNodeBuilder implements Builder<WebDevServerOptions> {
```

Comment:
Should it be called WebDevServerBuilder?

Reply:
Was this class inspired by the code in the CLI?

Reply:
yes, it should be. making the edit